File: 源码/EnjoySkin/app/src/main/java/com/enjoy/skin/MainActivity.java
package com.enjoy.skin;

import android.content.Context;
import android.content.Intent;
import android.content.pm.PackageInfo;
import android.content.pm.PackageManager;
import android.content.res.AssetManager;
import android.content.res.Resources;
import android.graphics.drawable.Drawable;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v4.view.ViewPager;
import android.support.v7.app.AppCompatActivity;
import android.util.AttributeSet;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.enjoy.skin.fragment.MusicFragment;
import com.enjoy.skin.fragment.RadioFragment;
import com.enjoy.skin.fragment.VideoFragment;
import com.enjoy.skin.widget.MyTabLayout;

import java.lang.reflect.Method;
import java.util.ArrayList;
import java.util.List;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        //控件的收集在这里完成
        setContentView(R.layout.activity_main); // 此时已经替换Factory2了

        View view = findViewById(R.id.test);

        MyTabLayout tabLayout = findViewById(R.id.tabLayout);
        ViewPager viewPager = findViewById(R.id.viewPager);
        List<Fragment> list = new ArrayList<>();
        list.add(new MusicFragment());
        list.add(new VideoFragment());
        list.add(new RadioFragment());
        List<String> listTitle = new ArrayList<>();
        listTitle.add("音乐");
        listTitle.add("视频");
        listTitle.add("电台");
        MyFragmentPagerAdapter myFragmentPagerAdapter = new MyFragmentPagerAdapter
                (getSupportFragmentManager(), list, listTitle);
        viewPager.setAdapter(myFragmentPagerAdapter);
        tabLayout.setupWithViewPager(viewPager);

    }


    /**
     * 进入换肤
     *
     * @param view
     */
    public void skinSelect(View view) {
        startActivity(new Intent(this, SkinActivity.class));
    }
}

File: 源码/gradle/demo/app/src/main/java/com/imooc/demo/MainActivity.kt
package com.imooc.demo

import android.content.pm.PackageManager
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.Toast
import com.imooc.biz.reading.Test
import java.lang.Exception

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        try {
            val appInfo = packageManager
                .getApplicationInfo(packageName,PackageManager.GET_META_DATA)
            val channelName = appInfo.metaData.getString("MTA_CHANNEL")
            Log.i("channel_test", "channel = $channelName")
        } catch (e: Exception) {
            // ignore
        }

        findViewById<View>(R.id.btn).setOnClickListener {

            Test().hello(it.context)

        }

        importantMethod()
        importantMethodTwo()
    }

    private fun importantMethod() {
    }

    private fun importantMethodTwo() {
        Log.i("test", "test")
    }
}